lib/asStream.ts

```diff
@@ -0,0 +1,50 @@
+import ObservableStore = require('.')
+
+import { Duplex as DuplexStream } from 'stream';
+
+class ObsStoreStream extends DuplexStream {
+
+  handler: (s: Record<string, unknown>) => void;
+
+  obsStore: ObservableStore;
+
+  constructor (obsStore: ObservableStore) {
+    super({
+      // pass values, not serializations
+      objectMode: true,
+    })
+    // dont buffer outgoing updates
+    this.resume()
+    // save handler so we can unsubscribe later
+    this.handler = (state: Record<string, unknown>) => this.push(state)
+    // subscribe to obsStore changes
+    this.obsStore = obsStore
+    this.obsStore.subscribe(this.handler)
+  }
+
+  // emit current state on new destination
+  pipe<T extends NodeJS.WritableStream>(dest: T, options?: { end?: boolean }): T {
+    let result = super.pipe(dest, options);
+    dest.write(this.obsStore.getState() as any)
+    return result;
+  }
+
+  // write from incomming stream to state
```

Comment:
This typo has irked me in the past!
```suggestion
  // write from incoming stream to state
```